### ISSUE TITLE
Joypad configuration won't read dpad

### ISSUE THREAD

**Issue by AlvaroBurnett** [LIJI32/SameBoy#96]
In Joypad -> Configure layout, dpad presses are not detected (it stays at "Press button for Right").

As a workaround, you can press Enter four times to skip the dpad configuration. This doesn't cause any trouble because the dpad works fine in the games.

**Comment by VladusD** [comment]
This still happens in Sameboy 0.12.1. I should add that I tried all buttons on my wireless Xbox One for Windows gamepad, and in addition to the d-pad, the analog triggers also seem to not be detected by the button layout configuration thing (don't know if it's a huge deal for gameboy, but still a bug I guess). 

And the workaround by @AlvaroBurnett works.

**Comment by anonymous7002** [comment]
Still in 0.13.6.

**Comment by ishax-kos** [comment]
And in 0.14.3.
I would guess that its a lack of hat switch support.

**Comment by TehMiloss** [comment]
Yup, same thing here, but I can't map it either to Analog or to D-Pad with my insideGadgets Wireless Gameboy Controller.
After plugging it in and turning on SameBoy - buttons A and B were swapped. I've tried to configure layout via Options, but SameBoy doesn't want to register D-Pad input at all. 
In Wireless GB/GBA Adapter D-Pad inputs are mapped to the Axis, so that may be the case, but there's no option to change it at all.

**Comment by LIJI32** [comment]
This doesn't seem to be the same issue, which was fixed in 0.15. It appears that the Wireless Gameboy Controller is doing some non-standard HID that is not compatible with SDL's controller library (and most likely JoyKit from the Cocoa frontend as well). I could have a look at it, but it doesn't seem like the controller's protocol is documented anywhere.

**Comment by LIJI32** [comment]
Looks like the insideGadgets Wireless Game Boy Controller supports mapping the D-Pad as buttons rather than an analog joystick if you hold the down button and while plugging the receiver in. Can you see if it solves the problem for you?
